### DISCUSSION TITLE
Docker packagerules do not get triggered

### DISCUSSION THREAD

**Discussion by giokara-work** [renovatebot/renovate#9064] [Request Help] [answered]
**What Renovate type, platform and version are you using?**

App: Hosted
Platform: Github
Version: 24.78.0

**Describe the bug**

Docker packagerules with daily schedule is never triggered.

**Relevant debug logs**

Relevant part in the renovate.json file:

```
    "hostRules": [
      {
        "baseUrl": "https://quay.io",
        "encrypted": {
          "password": "***********",
          "username": "***********"
        },
        "hostType": "docker"
      }
    ],
    "packageRules": [
      {
        "datasources": [
          "docker"
        ],
        "managers": [
          "dockerfile"
        ],
        "packagePatterns": [
          "^quay.io/our-org/tool-"
        ],
        "versioning": "semver",
        "schedule": [
          "after 1am"
        ],
        "groupName": "Docker containers",
        "enabled": true
      }
    ],
```

From the log of 10th of March, 2021, 1:05 am, no mention of the docker dependency is made, only of git-submodules which is also enabled:

```
INFO: Dependency extraction complete
{
  "baseBranch": "develop",
  "stats": {
    "managers": {
      "git-submodules": {
        "fileCount": 1,
        "depCount": 1
      }
    },
    "total": {
      "fileCount": 1,
      "depCount": 1
    }
  }
}
```

**Have you created a minimal reproduction repository?**

- [x] This is a really small bug, it does not need a reproduction (think small typo)
- [ ] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [ ] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

Possibly missing something in the configuration file?

**Comment by viceice** [comment, marked as answer]
What the name of your dockerfile? Renovate is using the following case-sensitive regex to match files:

https://github.com/renovatebot/renovate/blob/53ca91301f6a36c0c87e7d6b06b1bd8c9ec1b283/lib/manager/dockerfile/index.ts#L9

**Reply by giokara-work** [reply]
From the root of the project, the file is situated in `tooling/tool-version-sync/Dockerfile`.

**Reply by giokara-work** [reply]
If it is of any help, I tried this and no matches were returned:
```
✗ find /path/to/project -type f -regex '(^|/)Dockerfile\\.[^/]*$'
✗ find /path/to/project -type f -regex '(^|/|\\.)Dockerfile$'
```
Although the file is there:
```
✗ ls tooling/tool-version-sync/Dockerfile 
tooling/tool-version-sync/Dockerfile
```

**Reply by rarkins** [reply]
Please create a minimal reproduction repo (one package file, one dependency) and then post the debug logs from a run here. Renovate will log which file pattern it uses as well as info on how many files matched, etc.

**Reply by giokara-work** [reply]
Here is an example repository: https://github.com/oqton/test-renovate-docker.
The renovate logs now report:
```
INFO: Dependency extraction complete
{
  "baseBranch": "master",
  "stats": {
    "managers": {},
    "total": {
      "fileCount": 0,
      "depCount": 0
    }
  }
}
```

**Reply by viceice** [reply]
Your `includePaths` is not matching your `Dockerfile`. It makes an equals or minimatch compare.

https://github.com/renovatebot/renovate/blob/74d7691befa635c1220d8569936273bfb950e748/lib/workers/repository/extract/file-match.ts#L16

https://docs.renovatebot.com/configuration-options/#includepaths

**Reply by giokara-work** [reply]
Thanks, this gives already better results:
```
INFO: Dependency extraction complete
{
  "baseBranch": "master",
  "stats": {
    "managers": {
      "dockerfile": {
        "fileCount": 1,
        "depCount": 2
      }
    },
    "total": {
      "fileCount": 1,
      "depCount": 2
    }
  }
}
```

**Reply by giokara-work** [reply]
However at the moment that the docker updates should occur (it's scheduled to happen after 10 am), a separate error is thrown:
```
ERROR: Repository has unknown error
{
  "err": {
    "task": {
      "commands": [
        "rev-parse",
        "HEAD"
      ],
      "format": "utf-8"
    },
    "message": "fatal: Unable to read current working directory: No such file or directory\n",
    "stack": "Error: fatal: Unable to read current working directory: No such file or directory\n\n    at GitExecutorChain.onFatalException (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/runners/git-executor-chain.js:65:87)\n    at GitExecutorChain.<anonymous> (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/runners/git-executor-chain.js:56:28)\n    at Generator.throw (<anonymous>)\n    at rejected (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/runners/git-executor-chain.js:6:65)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)"
  }
}
```
The only change was the applied suggestion: https://github.com/oqton/test-renovate-docker/commit/51ce1625f7483e1842bfbd23bee4393fffb7ebd1. Any suggestions to resolve this?

**Reply by viceice** [reply]
please share more of the debug logs around the error

**Reply by giokara-work** [reply]
Seems to have been a temporary issue. The next run an hour later successfully ran. So the issue can be seen as resolved for us, many thanks!

**Reply by giokara-work** [reply]
For completeness, here is the entire log of the failed run:
```
INFO: Repository started
{
  "renovateVersion": "24.78.4"
}
DEBUG: Using localDir: /mnt/renovate/gh/oqton/test-renovate-docker
DEBUG: Repository cache is valid
DEBUG: initRepo("oqton/test-renovate-docker")
DEBUG: Overriding default GitHub endpoint
{
  "endpoint": "https://api.github.com/"
}
DEBUG: No dangling containers to remove
DEBUG: oqton/test-renovate-docker default branch = master
DEBUG: Using app token for git init
DEBUG: resetMemCache()
DEBUG: checkOnboarding()
DEBUG: isOnboarded()
DEBUG: findFile(renovate.json)
DEBUG: Initializing git repository into /mnt/renovate/gh/oqton/test-renovate-docker
DEBUG: git clone completed
{
  "durationMs": 673
}
ERROR: Repository has unknown error
{
  "err": {
    "task": {
      "commands": [
        "rev-parse",
        "HEAD"
      ],
      "format": "utf-8"
    },
    "message": "fatal: Unable to read current working directory: No such file or directory\n",
    "stack": "Error: fatal: Unable to read current working directory: No such file or directory\n\n    at GitExecutorChain.onFatalException (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/runners/git-executor-chain.js:65:87)\n    at GitExecutorChain.<anonymous> (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/runners/git-executor-chain.js:56:28)\n    at Generator.throw (<anonymous>)\n    at rejected (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/runners/git-executor-chain.js:6:65)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)"
  }
}
DEBUG: Unknown res
{
  "res": "unknown-error"
}
DEBUG: Repository timing splits (milliseconds)
{
  "splits": {},
  "total": 1698
}
DEBUG: http statistics
{
  "hostStats": [
    "api.github.com, 1 request, 236ms request average, 0ms queue average"
  ],
  "totalRequests": 1
}
INFO: Repository finished
{
  "durationMs": 1698
}
```

**Comment by giokara-work** [comment]
PRs are now being opened given the earlier feedback in our actual application (not the minimal example above). However some updates are still not being triggered. Excerpt of the renovate logs not finding updates for a Docker image:

![Screenshot 2021-04-02 at 17 33 07](https://user-images.githubusercontent.com/40686064/113430256-2225fa80-93da-11eb-812f-630dbee992c4.png)

Looking at the available versions in the Docker repository, there are versions to bump to in this case:

![Screenshot 2021-04-02 at 17 10 56](https://user-images.githubusercontent.com/40686064/113429665-3289a580-93d9-11eb-8c5f-2c44ccbafff4.png)

In the same renovate logs, for an exactly similar situation (same Docker repository and `packageRules`, different Docker image), update is done correctly:

![Screenshot 2021-04-02 at 17 33 20](https://user-images.githubusercontent.com/40686064/113430242-1afeec80-93da-11eb-9519-01e347710d80.png)

Available versions in the Docker repository:

![Screenshot 2021-04-02 at 17 38 58](https://user-images.githubusercontent.com/40686064/113430375-5c8f9780-93da-11eb-968a-e7d196f1fda9.png)

Would you have any hints on what can cause this selective updating?

**Reply by giokara-work** [reply]
Is there some maximum on the number of updates per `packageRules`? In our application, in theory, there might need to be up to 30 updates in the same renovate PR.

**Reply by rarkins** [reply]
This is a separate and unrelated topic of versioning systems. Please create a new discussion and give sufficient details there. Please copy/paste the logs and not screenshot them, so that they are easier for others.
